terraform/monitoring/03_service_slo.tf

```diff
@@ -0,0 +1,100 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Here we create a custom service that we attach our SLOs to
```

Comment:
Doesn't Service Monitoring detect our microservices automatically? I was under impression this is the case given we run GKE w/Istio

Reply:
This is true - originally I moved to a custom service because there isn't Terraform support in retrieving the services since Istio automatically detects/creates them.

However I realized we can use this string as an identifier:
ist:${var.project_id}-zone-us-central1-f-stackdriver-sandbox-default-frontend-external
This is a bit more "hardcoded" since we're manually creating the values instead of retrieving them from somewhere
Since we're not setting the zone when creating the project it will get automatically configured to us-central1-f

Let me know what approach you think is best:
1) Custom Service
2) Istio Service using string identifier

Reply:
Would it be useful in the end to setup some services via Custom Service with the rest using Istio in order to show customers both options?  Just like we are planning to have at least one service use custom metrics written via OT/OC for SLOs?

Reply:
I added a comment about the two approaches and why we chose the current approach. I definitely think it's possible to do what Brian suggests in the future. I'm closing the comment for now for this story, and we can discuss again about the future services.